tests/mock_crust/churn.rs

```diff
@@ -84,11 +84,24 @@ fn add_random_node<R: Rng>(rng: &mut R,
     let config = Config::with_contacts(&[nodes[proxy].handle.endpoint()]);
 
     nodes.insert(index, TestNode::builder(network).config(config).create());
-    if index <= proxy {
+    let (new_node, proxy) = if index <= proxy {
         (index, proxy + 1)
     } else {
         (index, proxy)
+    };
+
+    if len > (2 * min_section_size) {
+        let mut block_peer = gen_range_except(rng, 0, nodes.len(), Some(new_node));
+        while block_peer == proxy {
+            block_peer = gen_range_except(rng, 0, nodes.len(), Some(new_node));
+        }
```

Comment:
You can avoid the potentially unlimited number of random number generations:
```rust
let mut block_peer = gen_range_except(rng, 0, nodes.len() - 1, Some(new_node));
if block_peer >= proxy {
    block_peer += 1;
}
```
(Also below.)